### DISCUSSION TITLE
Ability to pass in a custom querybuilder similar to feathers-knex service adapter?

### DISCUSSION THREAD

**Discussion by evilbuck** [feathersjs-ecosystem/feathers-objection#161] [Ideas]
`feathers-knex` allows passing in a custom query builder via `params.knex`. is this something that is desirable to have feature parity?

Essentially it's the equivalent of 
```javascript
  _createQuery(params = {}) {
    const trx = params.transaction ? params.transaction.trx : null;
    const schema = params.schema || this.schema;

    // the only change is this line
    const query = params.knex ?? this.Model.query(trx);

    if (schema) {
      query.context({
        onBuild(builder) {
          builder.withSchema(schema);
        },
      });
    }
    return query;
  }
```

**Comment by dekelev** [comment]
There was never an attempt to achieve a feature parity with feathers-knex. These are 2 different libraries with different developers and different goals. Any PR can be accepted if it makes sense.

**Reply by evilbuck** [reply]
My apologies, I was mistaken. I think the readme￼ mentioned that it supported everything that feathers-knex supported plus objection features. I just assumed...
￼￼
I don't mind making a PR. I just wanted to make sure that this is a feature that would be worthwhile. 
I'll take a look and see if you have any contribution guidelines.￼￼

**Comment by dekelev** [comment]
There aren't any written guidelines for a PR, but you should add a reason to it, followed by usage code examples and eventually include relevant tests before it is merged.
